Console/Command/SqlMigrationShell.php

```diff
@@ -73,6 +73,23 @@ public function main() {
 
    } // End function main()
 
+   
+   /**
+     * Attempt to get a named mysql lock.
+     * Throws an exception on failure
+     * @return void
+     * @throws RuntimeException If unable to obtain a lock
+     */
+   public function getLock($timeout = 300) {
```

Comment:
if there is ever a migration that runs for more than 300 seconds, the instance waiting to acquire the lock will throw

Reply:
Aye. It's rare for us for have long-running migrations, but I am considering bumping this up to something conservative like 30/60/90 minutes. Removing the timeout entirely feels a little riskier (in terms of someone being notified about release failures) than having a long limit, but I'm open to opinions.

Reply:
It might be good to increase the default, but regardless it should probably be configurable when calling the script